src/napari_epitools/_widget.py

```diff
@@ -240,3 +247,173 @@ def run() -> tuple[npt.NDArray[np.float64], npt.NDArray[np.int64]]:
         )
 
     return run()
+
+
+def cell_statistics_widget() -> magicgui.widgets.Container:
+    """Create a widget for calculating cell statistics of labelled segmentations."""
+
+    widgets = _create_cell_statistics_widgets()
+    cell_statistics_widget = magicgui.widgets.Container(
+        widgets=widgets,
+        scrollable=False,
+    )
+
+    viewer = napari.current_viewer()
+    cell_statistics_widget.viewer = viewer
+
+    # Update cell_statistics when scrolling through frames
+    viewer.dims.events.current_step.connect(
+        lambda event: _update_cell_statistics(
+            layers=viewer.layers, frame=event.value[0]
+        ),
+    )
+
+    # calculate cell_statistics when pressing 'Run' button
+    cell_statistics_widget.run.changed.connect(
+        lambda: run_cell_statistics(
+            image=cell_statistics_widget.input_image.value,
+            labels=cell_statistics_widget.input_labels.value,
+        ),
+    )
+
+    # write the cell_statistics to CSV
+    cell_statistics_widget.export.changed.connect(
+        lambda: export_cell_statistics(
+            labels=cell_statistics_widget.input_labels.value,
+        ),
+    )
+
+    return cell_statistics_widget
+
+
+def _create_cell_statistics_widgets() -> list[Widget]:
+    """Create widgets for calculating and exporting cell statistics"""
+
+    image_tooltip = "Select an 'Image' layer to use for calculating cell statistics."
+    image = magicgui.widgets.create_widget(
+        annotation=napari.layers.Image,
+        name="input_image",
+        label="image",
+        options={"tooltip": image_tooltip},
+    )
+
+    labels_tooltip = (
+        "Select a 'Labels' layer to use for calculating cell statistics.\n"
+        "These should be the corresponding labels for the selected Image."
+    )
+    labels = magicgui.widgets.create_widget(
+        annotation=napari.layers.Labels,
+        name="input_labels",
+        label="labels",
+        options={"tooltip": labels_tooltip},
+    )
+
+    run_tooltip = "Calculate cell statistics for the selected Image and Labels"
+    run = magicgui.widgets.create_widget(
+        name="run",
+        label="Calculate statistics",
+        widget_type="PushButton",
+        options={"tooltip": run_tooltip},
+    )
+
+    export_tooltip = (
+        "Export the cell statistics for the selected Image and Labels to a CSV file"
+    )
+    export = magicgui.widgets.create_widget(
+        name="export",
+        label="Export statistics",
+        widget_type="PushButton",
+        options={"tooltip": export_tooltip},
+    )
+
+    return [image, labels, run, export]
+
+
+def _update_cell_statistics(
+    layers: list[napari.layers.Layer],
+    frame: int,
+) -> None:
+    """Update Labels cell_statistics for current frame"""
+
+    for layer in layers:
+        try:
+            layer.features = layer.metadata["cell_statistics"][frame]
+        except KeyError:
+            pass
+        except IndexError:
+            pass
```

Comment:
Should we log something here?

Reply:
Shall we set up logging in a separate pr?